### DISCUSSION TITLE
Binding to properties of other controls?

### DISCUSSION THREAD

**Discussion by mrlacey** [CommunityToolkit/Maui.Markup#151] [Q&A] [answered]
Is binding to properties of other controls supported by any of the `Bind` extensions?
I can't get it to work.

Ultimately I also want to be able to create instances of controls where some properties (& commands) are bound to a ViewModel (via the BindingContext) and some properties are bound to properties of other controls on the page.

I've been experimenting with a lot of different things that I thought should/might work but with no success.
Before I spend any more time on this, is it expected to be possible? (I'm probably just doing something silly.)

**Comment by TheCodeTraveler** [comment, marked as answer]
> Is binding to properties of other controls supported by any of the Bind extensions

Yup! That's known as a [View-to-View binding](https://learn.microsoft.com/dotnet/maui/xaml/fundamentals/data-binding-basics?view=net-maui-7.0#view-to-view-bindings).

Here's an example where I'm binding `CollectionView.HeightRequestProperty` to `SearchBar.HeightProperty`:
```cs
.Bind(CollectionView.HeightRequestProperty, SearchBar.HeightProperty.PropertyName, source: searchBar, convert: (double searchBarHeight) => this.Height - searchBarHeight)
```
https://github.com/ChilliCream/workshops/blob/e8c21a2c7b52ddd47ec641484272b42f825bb2f4/crypto-maui/frontend/X-final/Pages/ScreenerPage.cs#L26

**Reply by mrlacey** [reply]
Thanks. What I was doing was working but I'd actually encountered another issue. (Raising a MAUI bug for that separately.)
Your example also pointed me to the `Assign` method, which is really useful. I was creating instances as named variables and then adding them to the content of the page. This is much nicer. This is probably something that would benefit from improved documentation.

**Comment by CliffAgius** [comment]
Closed and answered
